make `test-Test` fails

```
make test-Test
    HINT The system image was recently rebuilt. Are you aware of the test-revise-* targets? See CONTRIBUTING.md. 
    JULIA test/Test
Test  (Worker) | Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)
Test       (1) |        started at 2020-09-19T19:11:50.776
Mock for testing retval of record(::DefaultTestSet, ::T <: Result) methods: Error During Test at An Error Mock:0

Mock for testing retval of record(::DefaultTestSet, ::T <: Result) methods: Test Failed at A Fail Mock:0
  Expression: 1
   Evaluated: 2
Stacktrace:
 [1] record(ts::Test.DefaultTestSet, t::Union{Test.Error, Test.Fail})
   @ Test ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:772
 [2] macro expansion
   @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:265 [inlined]
 [3] macro expansion
   @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113 [inlined]
 [4] top-level scope
   @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:259
@test_deprecated: Test Failed at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:807
  Expression: length(fails) == 2
   Evaluated: 0 == 2
Stacktrace:
 [1] top-level scope
   @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:807
 [2] top-level scope
   @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
 [3] top-level scope
   @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797
@test_deprecated: Error During Test at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:808
  Test threw exception
  Expression: fails[1] isa Test.LogTestFailure
  BoundsError: attempt to access 0-element Vector{Any} at index [1]
  Stacktrace:
   [1] getindex(A::Vector{Any}, i1::Int64)
     @ Base ./array.jl:809
   [2] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:808
   [3] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
   [4] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797
@test_deprecated: Error During Test at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:809
  Test threw exception
  Expression: fails[2] isa Test.LogTestFailure
  BoundsError: attempt to access 0-element Vector{Any} at index [2]
  Stacktrace:
   [1] getindex(A::Vector{Any}, i1::Int64)
     @ Base ./array.jl:809
   [2] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:809
   [3] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
   [4] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797
Test       (1) |         failed at 2020-09-19T19:12:28.600
Worker 1 failed running test Test:
Some tests did not pass: 326 passed, 1 failed, 2 errored, 14 broken.
Test: Test Failed at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:807
  Expression: length(fails) == 2
   Evaluated: 0 == 2
Stacktrace:
 [1] record(ts::Test.DefaultTestSet, t::Union{Test.Error, Test.Fail})
   @ Test ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:772
 [2] (::var"#23#32")()
   @ Main ~/julia-comparison/test/runtests.jl:346
 [3] cd(f::var"#23#32", dir::String)
   @ Base.Filesystem ./file.jl:104
 [4] top-level scope
   @ ~/julia-comparison/test/runtests.jl:81
 [5] include(mod::Module, _path::String)
   @ Base ./Base.jl:379
 [6] exec_options(opts::Base.JLOptions)
   @ Base ./client.jl:285
 [7] _start()
   @ Base ./client.jl:485
Test: Error During Test at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:808
  Test threw exception
  Expression: fails[1] isa Test.LogTestFailure
  BoundsError: attempt to access 0-element Vector{Any} at index [1]
  Stacktrace:
   [1] getindex(A::Vector{Any}, i1::Int64)
     @ Base ./array.jl:809
   [2] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:808
   [3] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
   [4] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797
Test: Error During Test at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:809
  Test threw exception
  Expression: fails[2] isa Test.LogTestFailure
  BoundsError: attempt to access 0-element Vector{Any} at index [2]
  Stacktrace:
   [1] getindex(A::Vector{Any}, i1::Int64)
     @ Base ./array.jl:809
   [2] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:809
   [3] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
   [4] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797

Test Summary: | Pass  Fail  Error  Broken  Total
  Overall     |  326     1      2      14    343
    Test      |  326     1      2      14    343
    FAILURE

The global RNG seed was 0xaf0456148bfd8783fb0aa0987da89921.

Error in testset Test:
Test Failed at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:807
  Expression: length(fails) == 2
   Evaluated: 0 == 2
Error in testset Test:
Error During Test at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:808
  Test threw exception
  Expression: fails[1] isa Test.LogTestFailure
  BoundsError: attempt to access 0-element Vector{Any} at index [1]
  Stacktrace:
   [1] getindex(A::Vector{Any}, i1::Int64)
     @ Base ./array.jl:809
   [2] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:808
   [3] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
   [4] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797
Error in testset Test:
Error During Test at /home/keno/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:809
  Test threw exception
  Expression: fails[2] isa Test.LogTestFailure
  BoundsError: attempt to access 0-element Vector{Any} at index [2]
  Stacktrace:
   [1] getindex(A::Vector{Any}, i1::Int64)
     @ Base ./array.jl:809
   [2] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:809
   [3] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1113
   [4] top-level scope
     @ ~/julia-comparison/usr/share/julia/stdlib/v1.6/Test/test/runtests.jl:797
ERROR: LoadError: Test run finished with errors
in expression starting at /home/keno/julia-comparison/test/runtests.jl:81
make[1]: *** [Makefile:25: Test] Error 1
make: *** [Makefile:580: test-Test] Error 2
```
The test passes when run in parallel with other tests though.

--- a/test/Makefile
+++ b/test/Makefile
@@ -23,11 +23,11 @@ default: all
 
 $(TESTS):
 	@cd $(SRCDIR) && \
-	$(call PRINT_JULIA, $(call spawn,$(JULIA_EXECUTABLE)) --check-bounds=yes --startup-file=no ./runtests.jl $@)
+	$(call PRINT_JULIA, $(call spawn,$(JULIA_EXECUTABLE)) --check-bounds=yes --startup-file=no --depwarn=error ./runtests.jl $@)
 
 $(addprefix revise-, $(TESTS)): revise-% :
 	@cd $(SRCDIR) && \
-    $(call PRINT_JULIA, $(call spawn,$(JULIA_EXECUTABLE)) --check-bounds=yes --startup-file=no ./runtests.jl --revise $*)
+    $(call PRINT_JULIA, $(call spawn,$(JULIA_EXECUTABLE)) --check-bounds=yes --startup-file=no --depwarn=error ./runtests.jl --revise $*)
 
 embedding:
 	@$(MAKE) -C $(SRCDIR)/$@ check $(EMBEDDING_ARGS)

